### ISSUE TITLE
Publish this information in a structured format

### ISSUE THREAD

**Issue by hjrb** [MicrosoftDocs/visualstudio-docs#2224]
Investigating this information in this format is not easy. It would be really helpful if you could publish this information in a machine readable format e.g. a database file or XML
Example: which workloads require and Android component to be installed? (Background: those fail download using the --layout parameter)

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b29a1009-5ef4-707b-258e-fe8bbe7275b6
* Version Independent ID: 00fac5a7-fc6d-c9cb-2d26-d820c7055890
* Content: [Visual Studio Enterprise 2017 workload and component IDs](https://docs.microsoft.com/en-us/visualstudio/install/workload-component-id-vs-enterprise?view=vs-2017)
* Content Source: [docs/install/workload-component-id-vs-enterprise.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/install/workload-component-id-vs-enterprise.md)
* Product: **visual-studio-dev15**
* GitHub Login: @TerryGLee
* Microsoft Alias: **tglee**

**Comment by TerryGLee** [comment]
@hjrb Thanks for your feedback; we'll take this under advisement.

**Comment by tristanbarcelon** [comment]
Would it be possible to post the WorkloadIDs and ComponentIDs per Visual Studio product in a file that is visible from github similar to what the dotnet team does for their [release index](https://dotnetcli.blob.core.windows.net/dotnet/release-metadata/releases-index.json) that is differentiated for each release (15.0, 15.9, 16.0, 16.1, etc.)? I'm hoping to create a friendly "bundler" utility that would allow me to define "must have" ComponentIDs while also only requiring workload IDs to be specified when installing. For example, I typically would like to install specific dotnetcore and .net framework targeting packs ComponentIDs which are either optional or recommended for WorkloadIDs that I am specifying. Arguably, this would be better than specifying all optional and recommended components be installed for WorkloadIDs I'm installing. By differentiating such a file by version, I can clearly tell what ComponentID has been added to a WorkloadID instead of having to visit a page. Even then, it's not clearly which release this table of WorkloadIDs and ComponentIDs is valid for. I've noticed that a minor version can have changes in ComponentIDs, such as what happened with TypeScript. Some versions ended up in the Unaffiliated list, while the current/newer TypeScript version is installed as required by its parent WorkloadID.
